ann_benchmarks/algorithms/tiledb/module.py

```diff
@@ -47,15 +66,18 @@ def fit(self, X):
         elif X.dtype == "float32":
             source_type = "F32BIN"
         maxtrain = min(50 * self._n_list, X.shape[0])
-        self.index = ingest(index_type="IVF_FLAT",
-                       array_uri=array_uri,
-                       source_uri=source_uri,
-                       source_type=source_type,
-                       size=X.shape[0],
-                       training_sample_size=maxtrain,
-                       partitions=self._n_list,
-                       input_vectors_per_work_item=100000000,
-                     )
+        self.index = ingest(
+            index_type="IVF_FLAT",
+            index_uri=array_uri,
+            source_uri=source_uri,
+            source_type=source_type,
+            size=X.shape[0],
+            training_sample_size=maxtrain,
+            partitions=self._n_list,
+            input_vectors_per_work_item=100000000,
+            mode=Mode.LOCAL
+        )
+        # memory_budget=-1 will load the data into main memory.
         self.index = IVFFlatIndex(uri=array_uri, dtype=X.dtype, memory_budget=-1)
```

Comment:
`dtype, memory_budget` args could be removed as this is the default behavior.

Reply:
Thanks, added this as a TODO and will fix this the next time I set up an EC2 instance and run this. Doing this b/c it seems nice to get this PR merged versus leaving in flux until then.